lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -1617,6 +1618,183 @@ FullApplySite SILCombiner::rewriteApplyCallee(FullApplySite apply,
+    return false;
+
+  SmallVector<ValueDecl *, 1> results;
+  // We'll let ObjectOutliner::replaceFindStringCall do the heavy lifting.
+  Builder.getASTContext().lookupInSwiftModule("_findStringSwitchCase", results);
+  if (results.size() != 1)
+    return false;
+
+  auto *funcDecl = dyn_cast<FuncDecl>(results.front());
+  if (!funcDecl)
+    return false;
+
+  std::string mangled = SILDeclRef(funcDecl, SILDeclRef::Kind::Func).mangle();
+  SILFunction *replacementFunc =
+      module.findFunction(mangled, SILLinkage::PublicExternal);
+  if (!replacementFunc)
+    return false;
+
+  results.pop_back();
+  Builder.getASTContext().lookupInSwiftModule("_allocateUninitializedArray",
+                                              results);
+  if (results.size() != 1)
+    return false;
+
+  funcDecl = dyn_cast<FuncDecl>(results.front());
+  if (!funcDecl)
+    return false;
+
+  mangled = SILDeclRef(funcDecl, SILDeclRef::Kind::Func).mangle();
+  // TODO: this only exists when there is an enum in the source file.
+  SILFunction *makeArray =
+      module.findFunction(mangled, SILLinkage::PublicExternal);
+  if (!makeArray)
+    return false;
+
+  results.pop_back();
+  // Look up "assert" to get the StaticString type.
+  Builder.getASTContext().lookupInSwiftModule("assert", results);
+  if (results.size() != 1)
+    return false;
+
+  funcDecl = dyn_cast<FuncDecl>(results.front());
+  if (!funcDecl)
+    return false;
+
+  mangled = SILDeclRef(funcDecl, SILDeclRef::Kind::Func).mangle();
+  SILFunction *assertFn =
+      module.findFunction(mangled, SILLinkage::PublicExternal);
+  if (!assertFn)
+    return false;
+
+  // General variables we'll need later.
+  auto dummyLoc = subject.getLoc();
+  auto wordTy = SILType::getBuiltinWordType(Builder.getASTContext());
+  auto byteTy = SILType::getBuiltinIntegerType(8, Builder.getASTContext());
+  auto boolTy = SILType::getBuiltinIntegerType(1, Builder.getASTContext());
+  auto i64Ty = SILType::getBuiltinIntegerType(64, Builder.getASTContext());
+  auto assertFnFuncTy = assertFn->getLoweredFunctionType();
+  auto staticStringBaseTy =
+      assertFnFuncTy->getParameters()[2].getSILStorageType(module,
+                                                           assertFnFuncTy);
+  auto staticStringTy = staticStringBaseTy.getObjectType();
+  auto staticStringAddrTy = staticStringBaseTy.getAddressType();
+  auto replacementFuncTy = replacementFunc->getLoweredFunctionType();
+  auto staticStringArrayTy = replacementFuncTy->getParameters()[0]
+                                 .getSILStorageType(module, replacementFuncTy)
+                                 .getAddressType();
+
+  // Move all the strings to the top of the function.
+  Builder.setInsertionPoint(stringLiterals.front());
+  for (auto *str : stringLiterals) {
+    str->moveBefore(stringLiterals.front());
+  }
+  // Set the insert point after the string literal instructions.
+  Builder.setInsertionPoint(stringCompares.front());
```

Comment:
Please create a separate builder for each insertion point. The purpose of a SILBuilder instance is to encapsulates information about current insertion point, like the debug scope. We're trying to get rid of the `setInsertionPoint` API--it's almost guaranteed to cause bugs.

Reply:
Good to know. Done.